### PR TITLE
Implement download function with tempfile

### PR DESCRIPTION
**Pull request by ThomDietrich**

### CODE REVIEW

**Comment by yann25** [comment]
I've tried it with my Sonoff S20 smart socket. It worked using my PI. 

With Windows there are permission difficulties. esptools cannot acces the tempfile even though I'm the only user and therefore have admin rights.

**Comment by ThomDietrich** [comment]
Sure it's permissions? Could you check if the file is deleted before? Shouldn't be but who knows. You can test by adding the delete parameter: `temp = tempfile.NamedTemporaryFile(delete=false)`

File: Sonoff-Tasmota_autoflash/tasmota_autoflash.py [hunk 1]
```diff
@@ -5,100 +5,70 @@
...
-    return ports
+API_URL = "https://api.github.com/repos/arendst/Sonoff-Tasmota/releases"
 
 def download():
-    #JSON Document url
-    url = "https://api.github.com/repos/arendst/Sonoff-Tasmota/releases"
-
-    #get data and parse it
-    response = urllib.urlopen(url)
+    temp = tempfile.NamedTemporaryFile(suffix='.sonoff-tasmota-firmare.bin')
```
Comment (yann25) [inline Sonoff-Tasmota_autoflash/tasmota_autoflash.py]: 'w' is missing

Comment (ThomDietrich) [inline Sonoff-Tasmota_autoflash/tasmota_autoflash.py]: That's default https://docs.python.org/2.7/library/tempfile.html#tempfile.NamedTemporaryFile

Comment (yann25) [inline Sonoff-Tasmota_autoflash/tasmota_autoflash.py]: I meant in .sonoff-tasmota.firmware.bin but it's not important

**Comment by yann25** [comment]
okay it works if you use `delete=False`

File: Sonoff-Tasmota_autoflash/tasmota_autoflash.py (hunk 1 above)
Comment (ThomDietrich) [inline Sonoff-Tasmota_autoflash/tasmota_autoflash.py]: oh, hahaha

**Comment by ThomDietrich** [comment]
Yeah okay but that should be solvable in a better way. delete=True will delete the file only after the program "think" it's not used any more

**Comment by yann25** [comment]
So what should do? Merge it or not?

**Comment by ThomDietrich** [comment]
Merge and fix

**Pull request merged**